nio_cli/commands/config.py

```diff
@@ -26,8 +27,58 @@ def config_project(name='.'):
                 tmp.write('PK_TOKEN={}\n'.format(pk_token))
             else:
                 tmp.write(line)
-        os.remove(env_location)
-        os.rename(tmp.name, env_location)
+        os.remove(conf_location)
+        os.rename(tmp.name, conf_location)
+    secure = input('Optional secure instance configuration [Y/N]: ')
+    if secure.lower() == 'y':
+        config_ssl(name, conf_location)
+
+# Assumes the user has OpenSSL already installed
+
+
+def config_ssl(name, conf_location):
+    ssl_cert = ''
+    ssl_key = ''
+
+    new_certs = input('Generate a self-signed certificate/key [Y/N]: ')
+
+    if (new_certs.lower() == 'y'):
+        subprocess.call('mkdir ssl &&  cd ssl', shell=True)
+        country = input('Enter two-letter country code: ')
+        state = input('Enter two-letter state code: ')
+        city = input('Enter city: ')
+        org = input('Enter name of organization: ')
+        owner = input('Enter name of owner: ')
+        user = input('Enter name of user: ')
+
+        gen_cert = ('openssl req -newkey rsa:2048 -nodes -keyout key.pem \
+                    -x509 -days 365 -out certificate.pem -subj \
+                    "/C={}/ST={}/L={}/O={}/OU=<{}/CN=localhost"').format(
+            country, state, city, org, owner, user)
+        subprocess.call(gen_cert, shell=True)
+        subprocess.call(
+            'openssl pkcs12 -inkey key.pem -in certificate.pem -export -out certificate.p12 -passout pass:', shell=True)
+        subprocess.call(
```

Comment:
Could these calls could be wrapped in a try/except block to catch when users don't have openssl installed?

Reply:
The error that gets thrown when attempting to run those commands makes it pretty clear that  openssl isn't installed.  Instead of the try/catch, I'll add a check at the beginning of the config_ssl() function to check if its installed